docs/docs/Tutorials/ROS2/PythonTemplate.md

```diff
@@ -0,0 +1,323 @@
+        self.declare_parameter('foo', self.policy.foo)
+        self.policy.foo = \
+            self.get_parameter('foo').get_parameter_value().double_value
+
+        # recompute any dependent variables
+        self.policy.update_params()
+```
+
+- Then, your control logic will go in the `target` function on line 36.
+    Modify the input args as well as the return value as necessary
+
+``` py linenums="36"
+    # Modify function inputs as desired
+    def target(self, *args, **kwargs):
+        '''Calculate target value from feedback inputs'''
+
+        # secret sauce
+
+        return 0.0  # obviously, modify to return proper target value
+```
+
+### Controller
+
+The `policy.target` function may be called from within the `Controller` class. You may call it inside any of
+the data callbacks to enable feedback control (for example):
+
+``` py
+    # To subscribe to any topic, simply define the specific callback, e.g. power_callback
+    def power_callback(self, data):
+        '''Callback for '/power_data' topic from Power Controller'''
+        # get target value from control policy
+        target_value = self.policy.target(data.rpm, data.scale, data.retract)
+
+        # send a command, e.g. winding current
+        self.send_pc_wind_curr_command(target_value, blocking=False)
+```
+
+Or, set up a loop in `main()` and run open-loop:
+
+``` py linenums="121"
+def main():
+    rclpy.init()
+    controller = Controller()
+    rate = controller.create_rate(50.0)  # Hz
+    while rclpy.ok():
+
+        # Open-loop control logic
+
+        rclpy.spin_once(controller)
+        rate.sleep()
+    rclpy.shutdown()
+```
+
+You may get feedback data from any of the buoy topics by simply creating a specific callback
+listed below. For feedback data you'd like to use in another area of the class, feel free to
+assign them to class variables.
+
+(Delete any callbacks you don't need in the `Controller` class)
+
+Available callback functions:
+
+`/ahrs_data` &rarr; `def ahrs_callback(self, data):`  
+`/battery_data` &rarr; `def battery_callback(self, data):`  
+`/spring_data` &rarr; `def spring_callback(self, data):`  
+`/power_data` &rarr; `def power_callback(self, data):`  
+`/trefoil_data` &rarr; `def trefoil_callback(self, data):`  
+`/powerbuoy_data` &rarr; `def powerbuoy_callback(self, data):`  
+
+You may also send commands from within the `Controller` class:
+
+`self.send_pump_command(duration_mins, blocking=False)`  
+`self.send_valve_command(duration_sec, blocking=False)`  
+`self.send_pc_wind_curr_command(wind_curr_amps, blocking=False)`  
+`self.send_pc_bias_curr_command(bias_curr_amps, blocking=False)`  
+`self.send_pc_scale_command(scale_factor, blocking=False)`  
+`self.send_pc_retract_command(retract_factor, blocking=False)`  
+
+---
+
+## Example
+
+An example using this interface will follow in the next tutorial: [Linear Damper Example (Python)](/Tutorials/ROS2/PythonLinearDamperExample.md)
```

Comment:
Is this page meant to be added in this PR? If not, this link will appear like a link that goes nowhere. It should be added when that page exists.